src/control/cmd/daos/pool.go

```diff
@@ -244,7 +247,34 @@ const (
 	dpiQueryAll     = C.uint64_t(^uint64(0)) // DPI_ALL is -1
 )
 
+func generateRankSet(ranklist *C.d_rank_list_t) string {
```

Comment:
FYI, there is already code to do this:
  * https://github.com/daos-stack/daos/blob/master/src/gurt/misc.c#L616 (create a `d_rank_range_list_t`)
  * https://github.com/daos-stack/daos/blob/master/src/gurt/misc.c#L662 (create a string to represent the range list)

I think that it would probably be better to use that code rather than this hand-rolled implementation. That having been said, I guess if this helper is only used for generating `system.RankSet`s and not for generating user-visible strings, it's fine. As noted in the other review comments, we are planning to get rid of requiring the intermediate string representation eventually.